openwisp_radius/saml/views.py

```diff
@@ -66,6 +67,22 @@ def post_login_hook(self, request, user, session_info):
         try:
             user.registered_user
         except ObjectDoesNotExist:
+            email = None
+            uid_is_email = 'email' in getattr(
+                settings, 'SAML_ATTRIBUTE_MAPPING', {}
+            ).get('uid', ())
+            if uid_is_email:
+                email = session_info['name_id'].text
+            if email is None:
+                email = session_info['ava'].get('email', [None])[0]
+            if email:
+                user.email = email
+                user.save()
```

Comment:
Saving this information without validating it is dangerous.

Here we need to run `full_clean()` and catch any possible exception in case the email we get from the identity provider is invalid for some reason. If the email is invalid wont save/create, log a warning and skip, eg:

```python
try:
    user.full_clean()
except ValidationError:
    logger.exception(f'Failed email validation for {user} during SAML user creation')
else:
    user.save()
    email_address = EmailAddress.objects.create(
       user=user, email=email, verified=True, primary=True
    )
```

Please add a new test for this error case, you can use `mock` or any other way to pass an invalid email and catch the validation error, there's similar tests in this and other modules you can base your test on, look for `ValidationError` in the test suites of the codebase.

Reply:
No problem.The new unified documentation looks great. I will make the suggested changes.